File: src/main/java/com/redbox/domain/funding/dto/ListResponse.java
package com.redbox.domain.funding.dto;

import com.redbox.domain.funding.entity.Funding;
import com.redbox.domain.funding.entity.FundingStatus;

import lombok.Getter;

import java.time.LocalDate;

@Getter
public class ListResponse {

    private Long fundingId; // 게시글 아이디

    private Long userId;
    private String userEmail;
    private String userName;

    private String fundingTitle;
    private String fundingContent;
    private int targetAmount;
    private int currentAmount;
    private double progressPercent;

    private FundingStatus fundingStatus;
    private String progress;

    private LocalDate fundingDate;
    private int fundingHits;
    private int fundingLikes;

    // Funding 엔티티를 매개변수로 받는 생성자 추가
    public ListResponse(Funding funding) {
        this.fundingId = funding.getFundingId();
        this.userId = funding.getUserId();
        this.userEmail = funding.getCreatedBy();
        this.userName = funding.getUserName();
        this.fundingTitle = funding.getFundingTitle();
        this.fundingContent = funding.getFundingContent();
        this.targetAmount = funding.getTargetAmount();
        this.currentAmount = funding.getCurrentAmount();
        this.progressPercent = ( (double) funding.getCurrentAmount() / funding.getTargetAmount()) * 100;
        this.progress = funding.getProgress().getText();
        this.fundingStatus = funding.getFundingStatus();
        this.fundingDate = funding.getFundingDate();
        this.fundingHits = funding.getFundingHits();
        this.fundingLikes = funding.getFundingLikes();
    }

}

File: migration/src/main/kotlin/com/redbox/domain/community/funding/dto/ListResponse.kt
package com.redbox.domain.community.funding.dto

import com.redbox.domain.community.funding.entity.Funding
import com.redbox.domain.community.funding.entity.FundingStatus
import java.time.LocalDate

data class ListResponse(

    val fundingId: Long?, // 게시글 아이디

    val userId: Long?,
    val userEmail: String?,
    //var userName: String,

    val fundingTitle: String?,
    val fundingContent: String?,
    val targetAmount: Int?,
    val currentAmount: Int?,
    val progressPercent: Double,

    val fundingStatus: FundingStatus,
    val progress: String?,

    val fundingDate: LocalDate?,
    val fundingHits: Int,
    val fundingLikes: Int,
) {
    constructor(funding: Funding) : this(
        fundingId = funding.fundingId,
        userId = funding.userId,
        userEmail = funding.createdBy,
        // userName
        fundingTitle = funding.fundingTitle,
        fundingContent = funding.fundingContent,
        targetAmount = funding.targetAmount,
        currentAmount = funding.currentAmount,
        progressPercent = (funding.targetAmount?.let { funding.currentAmount?.toDouble()?.div(it.toDouble()) })?.times(100) ?: 1.0,
        fundingStatus = funding.fundingStatus,
        progress = funding.progress.text,
        fundingDate = funding.fundingDate,
        fundingHits = funding.fundingHits,
        fundingLikes = funding.fundingLikes
    )
}